.github/workflows/publish-and-release.yml

```diff
@@ -7,130 +7,151 @@ jobs:
+      with:
+        run: git log --format="* %s" ${{ steps.get-latest-tag.outputs.tag }}..HEAD | { grep -v "(IGNORE)" || :; }
+
+    - name: Log version & changelog
+      run: |
+        echo "Version: $VERSION"
+        echo "Version tag: $VERSION_TAG"
+        echo "Latest tag detected: $LATEST_TAG"
+        echo "Changelog: $CHANGELOG"
+      env:
+        VERSION: ${{ steps.set-version.outputs.version }}
+        VERSION_TAG: ${{ steps.set-version.outputs.version_tag }}
+        LATEST_TAG: ${{ steps.get-latest-tag.outputs.tag }}
+        CHANGELOG: ${{ steps.changelog.outputs.stdout }}
+
+    - name: Prevent empty release
+      if: ${{ steps.changelog.outputs.stdout == '' }}
+      uses: actions/github-script@v3
+      with:
+        script: |
+          core.setFailed("No changes since prior release")
+
+    - name: Update changelog
+      run: |
+        (head -8 CHANGELOG.md && echo "## $VERSION" && date "+_%d %B %Y_" && echo "" && (echo "$CHANGELOG" | sed -E 's_\(#([0-9]+)\)_([#\1](https://github.com/contentauth/c2pa-rs/pull/\1)\)_') && tail -n +9 CHANGELOG.md) > CHANGELOG.new.md
+        mv CHANGELOG.new.md CHANGELOG.md
+      env:
+        VERSION: ${{ steps.set-version.outputs.version }}
+        CHANGELOG: ${{ steps.changelog.outputs.stdout }}
+
+    - name: Install Rust toolchain
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+
+    - name: Bump crate versions
+      run: |
+        sed -i "s/^version = \"[^\"]*\"$/version = \"$VERSION\"/;" Cargo.toml
+      env:
+        VERSION: ${{ steps.set-version.outputs.version }}
+
+    - name: Report differences for "prepare (release)" commit
+      run: git diff
+
+    - name: Commit Cargo.toml and changelog
+      uses: stefanzweifel/git-auto-commit-action@v4
+      id: commit
+      with:
+        commit_message: Prepare ${{ steps.set-version.outputs.version }} release
+        commit_user_name: Adobe CAI Team
+        commit_user_email: noreply@adobe.com
+
   release-crate:
     name: Release c2patool Rust crate
     needs: repo-prep
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v3
-      - name: Create release
-        uses: ncipollo/release-action@v1
-        with:
-          body: ${{  needs.repo-prep.outputs.changelog-body }}
-          commit: ${{ needs.repo-prep.outputs.commit-hash }}
-          prerelease: true # remove at 1.0
-          tag: ${{  needs.repo-prep.outputs.latest-tag }}
-          token: ${{ secrets.GH_ADMIN_COMMIT_TOKEN }}
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: Create release
+      uses: ncipollo/release-action@v1
+      with:
+        body: ${{ needs.repo-prep.outputs.changelog-body }}
+        commit: ${{ needs.repo-prep.outputs.commit-hash }}
+        prerelease: true # remove at 1.0
+        tag: ${{ needs.repo-prep.outputs.latest-tag }}
+        token: ${{ secrets.GH_ADMIN_COMMIT_TOKEN }}
+
     - name: Publish crate
```

Comment:
I think this is where the problem was; the indentation was inconsistent for the `steps` array above.